sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -296,8 +296,10 @@ case class LoadDataCommand(
           s"do not match number of partitioned columns in table " +
           s"(${targetTable.partitionColumnNames.size})")
       }
+      val resolver = sparkSession.sessionState.conf.resolver
       partition.get.keys.foreach { colName =>
-        if (!targetTable.partitionColumnNames.contains(colName)) {
+        if (!targetTable
+          .partitionColumnNames.exists(fieldName => resolver(fieldName, colName))) {
```

Comment:
Please see https://github.com/apache/spark/pull/24903#discussion_r303724838 . 
We need to change more~

Reply:
done